modules/transaction-payment/src/lib.rs

```diff
@@ -855,68 +913,55 @@ where
 	) -> Result<(T::AccountId, Balance), DispatchError> {
 		match call.is_sub_type() {
 			Some(Call::with_fee_path { fee_swap_path, .. }) => {
+				// pre check before set OverrideChargeFeeMethod
 				ensure!(
-					fee_swap_path.len() > 1
+					fee_swap_path.len() <= T::TradingPathLimit::get() as usize
+						&& fee_swap_path.len() > 1
 						&& fee_swap_path.first() != Some(&T::NativeCurrencyId::get())
 						&& fee_swap_path.last() == Some(&T::NativeCurrencyId::get()),
 					Error::<T>::InvalidSwapPath
 				);
+
+				let fee_aggregated_path: Vec<AggregatedSwapPath<CurrencyId>> =
+					vec![AggregatedSwapPath::<CurrencyId>::Dex(fee_swap_path.clone())];
+
+				// put in storage after check
+				OverrideChargeFeeMethod::<T>::put(ChargeFeeMethod::FeeAggregatedPath(fee_aggregated_path.clone()));
```

Comment:
```suggestion
				OverrideChargeFeeMethod::<T>::put(ChargeFeeMethod::FeeAggregatedPath(&fee_aggregated_path));
```

Reply:
`fee_aggregated_path` is reference here, change it to `fee_aggregated_path.to_vec()` according to complier suggest